src/xunit.console/Visitors/DiagnosticMessageVisitor.cs

```diff
@@ -18,21 +18,35 @@ public DiagnosticMessageVisitor(object consoleLock, string assemblyDisplayName,
             this.noColor = noColor;
         }
 
-        protected override bool Visit(IDiagnosticMessage diagnosticMessage)
+        /// <inheritdoc/>
+        public override bool OnMessage(IMessageSinkMessage message)
         {
             if (showDiagnostics)
```

Comment:
Note: I moved casting and visitation inside of the if (showDiagnostics) test. This might change observed behavior/bugs for someone... let alone from the fact that subclasses will no longer see any messages except IDiagnosticMessages.